actioncable/lib/action_cable/connection/base.rb

```diff
@@ -272,6 +293,27 @@ def successful_request_message
             env["REQUEST_METHOD"], env["HTTP_CONNECTION"], env["HTTP_UPGRADE"]
           ]
         end
+
+        def expects_pong_response_to_heartbeat?
+          protocol&.start_with?("actioncable-v1.1-")
+        end
+
+        def half_open?
+          @last_message_received_at.before?(HALF_OPEN_CONNECTION_TIMEOUT.ago)
+        end
+
+        def process_pong(payload)
+          latency = Time.now.to_f - payload["message"].to_f
+
+          if latency.positive?
+            ActiveSupport::Notifications.instrument(
+              "connection.latency",
+              value: latency,
+              connection_identifier: connection_identifier,
+              identifiers: identifiers.index_with { |id| instance_variable_get("@#{id}") }
+            )
+          end
```

Comment:
I'm on the fence about this, maybe it would be better to expose this in some other way to the app?

Reply:
Ok. I'll leave this as is for now then.